src/librbd/api/Mirror.cc

```diff
@@ -1973,8 +1973,8 @@ int Mirror<I>::image_info_list(
     map<std::string, cls::rbd::MirrorImageStatus> statuses;
 
     int r = librbd::cls_client::mirror_image_status_list(&io_ctx, last_read,
-                                                         max, &images,
-                                                         &statuses);
+                                                         max - entries->size(),
+                                                         &images, &statuses);
```

Comment:
@dillaman The retrieved images list is filtered below in the `for` loop and may result in `entries->size()` being still smaller than `max` and yet another cls call. May be check for `entries->size()` in the `for` loop below and break when the max is reached instead?

Reply:
Swapped